### ISSUE TITLE
When deploying a program, add option to auto extend to correct size

### ISSUE THREAD

**Issue by jacobcreech** [anza-xyz/agave#564]
#### Problem
When you deploy a program, it initially takes up 2x the program size. At a later date if you try to deploy larger than the currently allocated size, you get the following error:

```
Error: Deploying program failed: Error processing Instruction 0: account data too small for instruction
```
The correct way to solve this is to use `solana program extend <PROGRAM_ID> <MORE_BYTES>` to extend the program, then deploy it again. 

Developers attempting to calculate this amount over and over again have become frustrated and some have just spent the money to allocate way more than they should. Some even as much as 10mb. This is bad both for the cluster and developer experience.

#### Proposed Solution
When deploying a program with a binary that is too big for the allocated amount onchain, have two options for people to automatically extend to the amount required for their binary:
1. Suggest the command `solana program extend <program_id> <more_bytes>` with the options filled in instead of the current `account data too small for instruction` error. This will allow the devs to just copy paste and run the command.
2. Have a flag on the `solana program deploy` cli that automatically extends if a deployment requires it. Suggestion: `solana program deploy <program_id> --auto-extend`

**Comment by defi0x1** [comment]
Is possible to take this issue?

**Comment by jacobcreech** [comment]
> Is possible to take this issue?

You don't need to ask, PRs are welcome! Feel free to ask any clarifying questions.

**Comment by Nagaprasadvr** [comment]
I was looking at the code base to see where do we allocate 2x space for the program data account couldn't find any instance can anyone help me with this ..

**Comment by jacobcreech** [comment]
Fixed in #791 🎉
